src/App.tsx

```diff
@@ -1,61 +1,104 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { getNewPostId } from './utils/todoUtiles';
 
 export const App = () => {
+  const [currentTodos, setCurrentTodos] = useState(todosFromServer);
+  const [newTitle, setNewTitle] = useState('');
+  const [titleValidation, setTitleValidation] = useState(false);
+  const [newUserId, setNewUserId] = useState(0);
+  const [userValidation, setUserValidation] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const input = event.target.value;
+
+    setTitleValidation(!input.trim());
+    setNewTitle(input);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const input = event.target.value;
+
+    setUserValidation(!input);
+    setNewUserId(+input);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setTitleValidation(!newTitle.trim());
+    setUserValidation(!newUserId);
+
+    if (!newTitle.trim() || !newUserId) {
+      return;
+    }
+
+    const newTodo = {
+      id: getNewPostId(currentTodos),
+      title: newTitle,
+      completed: false,
+      userId: newUserId,
+    };
+
+    const newTodoList = [...currentTodos, newTodo];
+
+    setCurrentTodos(newTodoList);
```

Comment:
```suggestion
    setCurrentTodos(curr => [...curr, newTodo]);
```
always use callback for setting state which is based on the previous one